Backend/ManagerLayer/Login/UserLoginManager.cs

```diff
@@ -43,52 +42,30 @@ public class UserLoginManager
             }
             ////////////////////////////////////////
             
-            _userManagementManager = new UserManagementManager(_db);
-            var user = _userManagementManager.GetUser(ssoID);
+            _userManagementManager = new UserManagementManager();
+            var user = _userManagementManager.GetUser(_db, ssoID);
             // check if user does not exist
             if (user == null)
             {
                 // create new user
-                try
-                {
-                    user = _userManagementManager.CreateUser(Username, ssoID);
-                    _db.SaveChanges();
-                    newLog = new LogRequestDTO(ssoID.ToString(), Username,
-                        "Login/Registration API", user.Username, "Successful registration of new User", 
-                        "Line 54 UserLoginManager in ManagerLayer\n" +
-                        "Route Reference UserController in WebApi-PointMap");
-                    loggingManager.sendLogSync(newLog);
-
-                }
-                catch (InvalidEmailException ex)
-                {
-                    throw new InvalidEmailException(ex.Message);
-                }
-                catch (InvalidDbOperationException)
-                {
-                    _db.Entry(user).State = System.Data.Entity.EntityState.Detached;
-                    throw new InvalidDbOperationException("User was not created");
-                }
+                user = _userManagementManager.CreateUser(_db, Username, ssoID);
+                newLog = new LogRequestDTO(ssoID.ToString(), Username,
+                    "Login/Registration API", user.Username, "Successful registration of new User", 
+                    "Line 51 UserLoginManager in ManagerLayer\n" +
+                    "Route Reference UserController in WebApi-PointMap");
+                loggingManager.sendLogSync(newLog);
             }
             _authorizationManager = new AuthorizationManager();
             Session session = _authorizationManager.CreateSession(_db, user);
-            try
-            {
-                _db.SaveChanges();
-            }
-            catch (Exception)
-            {
-                _db.Entry(session).State = System.Data.Entity.EntityState.Detached;
-                throw new InvalidDbOperationException("Session was not created");
-            }
-            LoginManagerResponseDTO response;
-            response = new LoginManagerResponseDTO
+            _db.SaveChanges();

```

Comment:
Move to controller, make sure all errors have a chance to be caught before saving changes